lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -226,20 +236,25 @@ def log.trace?
         rescue KubeException => kube_error
           raise Fluent::ConfigError, "Invalid Kubernetes API #{@apiVersion} endpoint #{@kubernetes_url}: #{kube_error.message}"
         end
+        if @checkpoint_enabled
+          @checkpoint_thread_running = true
```

Comment:
Shouldnt this come after the the thread initialization in the following lines? maybe need some sort of mutex to ensure it happens only once?

Reply:
@jcantrill Will this not be executed only once by default? (please ignore my current flawed implementation), but I don't get the requirement for making sure that `thread_running` is set to true only once, I think I am missing the user case for this.

Reply:
I defer to your findings here as I was not the original author of this plugin and was making a casual observation.  I would expect initialization only to occur again if say the process was restarted.  Alternatively if fluent properly supported multi-threading this could additionally be an issue

Reply:
From what I have gone through other plugins, they all make the same assumption of just one thread having this. So I will remove the mutex in the code. As for the initialization happening multiple times, the init function is idempotent.